src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas2/ApplicationService.kt

```diff
@@ -52,16 +52,22 @@ class ApplicationService(
   @Value("\${url-templates.frontend.cas2.submitted-application-overview}") private val submittedApplicationUrlTemplate: String,
 ) {
 
-  fun getAllApplicationsForUser(user: NomisUserEntity): List<Cas2ApplicationSummary> {
-    return applicationRepository.findAllCas2ApplicationSummariesCreatedByUser(user.id)
+  fun getAllApplicationsForUser(user: NomisUserEntity, pageCriteria: PageCriteria<String>): Pair<MutableList<Cas2ApplicationSummary>, PaginationMetadata?> {
```

Comment:
Can we squash this commit into the previous one? Ideally we'd have all tests passing on each commit, and I don't think the first commit works without this change.

Reply:
Ahh yes, hadn't thought of structuring the commits in a way that tests always pass after very commit.  Good reasoning.

Have done this.